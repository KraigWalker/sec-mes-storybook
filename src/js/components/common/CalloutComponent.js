'use restrict';
import React from 'react';

class CalloutComponent extends React.Component {
 componentWillMount() {
	}
	componentDidMount() {

	}
	componentWillReceiveProps(newProps) {

	}
	shouldComponentUpdate(newProps, newState) {
		return true;
	}
	componentWillUpdate(nextProps, nextState) {

	}
	render() {
		return (
			<div className={this.props.dClass} onClick={this.props.selectComponent}>
				<p className={this.props.pClass}>{this.props.paraText}</p>
			</div>
		);
	}
}
CalloutComponent.propTypes = {
	dClass: React.PropTypes.string,
	selectComponent: React.PropTypes.func,
	pclass: React.PropTypes.string,
	paraText: React.PropTypes.string,
}
CalloutComponent.defaultProps = {
	dClass: 'callout',
	pClass: '',
	selectComponent: function(e) {return null},
	paraText: 'Contrary to popular belief, Lorem Ipsum is not random text. It has roots in a piece of classical Latin literature from 45 BC, making it 2000 years old.',
}

export default CalloutComponent;