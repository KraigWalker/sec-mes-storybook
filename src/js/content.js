
const content = {
	DYB: {
		landingPageTitle: 'Messages',
		landingPageMessage: 'Want to talk to the team at B about your account? Safely share your details in a secure message to get all the help you need.',
		inboxPageTitle: 'Your secure message',
		sentPageTitle: 'Your sent message',
		newMessagePageTitle: 'New secure message',
		replyMessageTitle: 'Reply',
		back: 'Back',
		send: 'Send message',
		maxCharLimit: 'Maximum character limit is three thousand',
		message: 'Message',
		messages: 'Secure messages',
		messageRelatesTo: 'What account’s it about?',
		subject: 'Subject',
		messageSent: 'Message sent',
		messageDeleted: 'Message deleted',
		ok: 'OK',
		draftBody: 'Message saved as draft',
		leaveMessageHeading: 'Oops, your message hasn’t sent yet',
		returnToMessage: 'Continue with message',
		saveDraft: 'Save draft',
		leaveMessageBody: 'Sure you want to leave this page? If you do, your message won’t be sent or saved as a draft.',
		charLeft: 'Characters Left:',
		sorryHeader: 'Something’s gone wrong',
		tryAgain: 'Hmm. Looks like there’s been a technical problem. Why not give it another go?',
		getInTouch: 'Still not working? Try another way to get in touch with the team at B.',
		retry: 'Try again',
		leavePage: 'Leave page',
		accError: 'Oops. You’ll have to choose what account your message is about. General enquiry? Just choose ‘No specific account’.',
		subError: 'Oops. You’ll have to choose a subject for your secure message. ',
		newSecureMessage: 'New secure message',
		thatsallTextInbox: 'That’s all of them. Any new messages the team at B send you will show at the top of your inbox. Remember that only messages from the last 13 months are shown here.',
		thatsallTextDraft: 'That’s all of them. Any new messages you save will show at the top of your drafts. Remember that only messages from the last 13 months are shown here.',
		thatsallTextSend: 'That’s all of them. Any new messages you send will show at the top of your sent items. Remember that only messages from the last 13 months are shown here.',
		noInboxMessages: 'Hmm, looks like you haven’t been sent any secure messages. Remember that only messages from the last 13 months are shown here.',
		noSentMessages: 'Hmm, looks like you haven’t sent any secure messages. Remember that only messages from the last 13 months are shown here.',
		noDraftMessages: 'Hmm, looks like you haven’t saved any secure messages. Remember that only messages from the last 13 months are shown here.',
		deleteMessageHeading: 'Sure you want to delete this message?',
		deleteMessageBody: 'If you delete this message now, you won’t be able to get it back.',
		dontDelButton: 'No, don’t delete',
		delButton: 'Yes, delete',
		messageVal: 'Oops. Looks like your message has gone over the character limit. Try and cut it down a little then give it another go',
		showMore: 'Show more messages',
		backToAccounts: 'Back to accounts',
		faqLink: 'You’ll see all your secure messages from the last 13 months here. For more info about your message history, check out the FAQs in the help section.'
	},

	CB: {
		landingPageTitle: 'Messages',
		landingPageMessage: 'If you want to talk about something specific to you or your account, you can share your details in a secure message. It’s safe and simple, and you’ll get all the help you need.',
		inboxPageTitle: 'Your secure message',
		sentPageTitle: 'Your sent message',
		newMessagePageTitle: 'New secure message',
		replyMessageTitle: 'Reply',
		back: 'Back',
		send: 'Send message',
		maxCharLimit: 'Maximum character limit is three thousand',
		message: 'Message',
		messages: 'Secure messages',
		messageRelatesTo: 'Related account',
		subject: 'Subject',
		messageSent: 'Message sent',
		messageDeleted: 'Message deleted',
		ok: 'OK',
		draftBody: 'Message saved as draft',
		leaveMessageHeading: 'Your message hasn’t been sent yet',
		returnToMessage: 'Continue with message',
		saveDraft: 'Save draft',
		leaveMessageBody: 'If you leave this page now, your message won’t be sent or saved as a draft.',
		charLeft: 'Characters Left:',
		sorryHeader: 'Sorry, there’s been a technical problem',
		tryAgain: 'It looks like something’s gone wrong in the background. Please try again.',
		getInTouch: 'If you’re still having problems, please use another method to get in touch with us.',
		retry: 'Try again',
		leavePage: 'Leave page',
		accError: 'Please select which account your message relates to. If it’s a general enquiry, select ‘No specific account’.',
		subError: 'Please select a subject for your secure message.',
		newSecureMessage: 'New secure message',
		thatsallTextInbox: 'That’s all of them. Any new messages we send you will appear at the top of your inbox. Please note that only messages from the last 13 months are displayed here.',
		thatsallTextDraft: 'That’s all of them. Any new messages you save will appear at the top of your drafts. Please note that only messages from the last 13 months are displayed here.',
		thatsallTextSend: 'That’s all of them. Any new messages you send will appear at the top of your sent items. Please note that only messages from the last 13 months are displayed here.',
		noInboxMessages: 'You haven’t received any secure messages. Please note that only messages from the last 13 months are displayed here.',
		noSentMessages: 'You haven’t sent any secure messages. Please note that only messages from the last 13 months are displayed here.',
		noDraftMessages: 'You haven’t saved any secure messages. Please note that only messages from the last 13 months are displayed here.',
		deleteMessageHeading: 'Are you sure you want to delete this message?',
		deleteMessageBody: 'If you delete this message now, you won’t be able to recover it.',
		dontDelButton: 'No, don’t delete',
		delButton: 'Yes, delete',
		messageVal: 'Your message has exceeded the maximum character limit. Please reduce the length of your message then try and send it again.',
		showMore: 'Show more messages',
		backToAccounts: 'Back to accounts',
		faqLink: 'You’ll see all your secure messages from the last 13 months here. For more information about your message history, please visit the FAQs in the help section.',
	},
};

export default content;
