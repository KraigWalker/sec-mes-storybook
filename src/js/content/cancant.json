{   "scope_name":"voice",
    "scope_description":"Voice",
    "consent_format":"DEFAULT",
    "inclusions":[
        {"description":"How much you've got in your account","screen_order":1},
        {"description":"When your bills are due","screen_order":2},
        {"description":"When and what you have been paid","screen_order":3}
    ],
    "exclusions":[
        {"description":"Make any payments","screen_order":1},
        {"description":"Make any transfers","screen_order":2},
        {"description":"Access any other accounts","screen_order":3}
    ]
}