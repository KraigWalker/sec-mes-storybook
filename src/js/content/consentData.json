{
    "user_id" : "1000000123",
    "consents" : [
       {
          "consent_id" : "44458765-a654drt-76gt-8990h-98iuh",
          "client_id" : "877ghf-ytrfdt-76ft-jhgt-765",
          "client_name" : "Amazon",
          "scope_name" : "Voice",
          "scope_description" : "Voice",
          "consent_format" : "Default",
          "required_trust_level" : "40",
          "start_date" : "2017-09-22T14:01:09Z",
          "expiry_date" : "2017-12-22T14:01:09Z",
          "client_scope_credentials" : [
             {
                "credential_type" : "pin",
                "pin_type" : "alexa_pin"
             }
        
          ],
          "client_scope_tandcs_applicable" : "true",
          "client_scope_tandcs_service_name" : "service_name",
          "must_accept_new_client_scope_tandcs" : "true",
          "thumbnail_link" : "<url to third party logo>",
          "accounts" : [
             {
                "account_number" : "121212-19857624",
                "account_name" : "B Current",
                "nickname" : "My B",
                "product_type" : "B"
             }
             
          ]
       },
       {
        "consent_id" : "44458765-a654drt-76gt-8990h-98iuh",
        "client_id" : "877ghf-ytrfdt-76ft-jhgt-765",
        "client_name" : "Amazon",
        "scope_name" : "Payments",
        "scope_description" : "Payments",
        "consent_format" : "Default",
        "required_trust_level" : "40",
        "start_date" : "2017-09-22T14:01:09Z",
        "expiry_date" : "2017-12-22T14:01:09Z",
        "client_scope_credentials" : [
          
        ],
        "client_scope_tandcs_applicable" : "true",
        "client_scope_tandcs_service_name" : "service_name",
        "must_accept_new_client_scope_tandcs" : "false",
        "thumbnail_link" : "<url to third party logo>",
        "accounts" : [
           {
              "account_number" : "242424-19857111",
              "account_name" : "B Current",
              "nickname" : "My B",
              "product_type" : "B"
           }
           
        ]
     }
    ]
 }