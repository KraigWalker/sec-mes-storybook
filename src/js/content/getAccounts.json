{
	"accounts": [{
		"type": "current",
		"number": "824404-20012550",
		"product": {
			"code": "136",
			"name": "B Current",
			"description": "DYB CURRENT ACCOUNT"
		},
		"actions_available": {
			"/account/alerts": true,
			"/account/consent": false,
			"/account/payments/transfer/in": true,
			"/account/mandates/so/write": true,
			"/account/statements/order": true,
			"/account/mandates/so/read": true,
			"/account/books/cheque/reorder": true,
			"/account/transactions/readwrite": false,
			"/account/sweeps": true,
			"/account/beneficiaries/read": true,
			"/account/beneficiaries/delete": true,
			"/account/sweeps/transfer/out": false,
			"/account/beneficiaries/biller/write": true,
			"/account/read": true,
			"/account/projections": true,
			"/account/payments/uk/ics/in": false,
			"/account/books/payin/reorder": true,
			"/account/beneficiaries/account/readwrite": true,
			"/account/mandates/dd/read": true,
			"/account/preferences/paperstatementoptout": true,
			"/account/statements/read": true,
			"/account/payments/uk/default/out": true,
			"/account/payments/transfer/out": true,
			"/account/transactions/write": true,
			"/account/transactions/read": true,
			"/account/pots": false,
			"/account/mandates/dd/write": true
		},
		"bank_id": "CB",
		"metadata": {
			"display_name": null,
			"display_order": null
		},
		"balances": [{
			"type": "current",
			"amount": {
				"value": 2914.00,
				"currency": "GBP"
			}
		}, {
			"type": "available",
			"amount": {
				"value": 2914.00,
				"currency": "GBP"
			}
		}],
		"id": "92ff9258-f639-43c6-a58c-bacef915667c"
	}, {
		"type": "savings",
		"number": "824404-20012899",
		"product": {
			"code": "138",
			"name": "B Instant Savings",
			"description": "DYB SAVINGS ACCOUNT"
		},
		"actions_available": {
			"/account/alerts": false,
			"/account/consent": false,
			"/account/payments/transfer/in": true,
			"/account/mandates/so/write": false,
			"/account/statements/order": true,
			"/account/mandates/so/read": false,
			"/account/books/cheque/reorder": false,
			"/account/transactions/readwrite": false,
			"/account/sweeps": false,
			"/account/beneficiaries/read": false,
			"/account/beneficiaries/delete": false,
			"/account/sweeps/transfer/out": true,
			"/account/beneficiaries/biller/write": false,
			"/account/read": true,
			"/account/projections": false,
			"/account/payments/uk/ics/in": false,
			"/account/books/payin/reorder": false,
			"/account/beneficiaries/account/readwrite": false,
			"/account/mandates/dd/read": false,
			"/account/preferences/paperstatementoptout": true,
			"/account/statements/read": true,
			"/account/payments/uk/default/out": false,
			"/account/payments/transfer/out": true,
			"/account/transactions/write": true,
			"/account/transactions/read": true,
			"/account/pots": true,
			"/account/mandates/dd/write": false
		},
		"bank_id": "CB",
		"metadata": {
			"display_name": null,
			"display_order": null
		},
		"balances": [{
			"type": "current",
			"amount": {
				"value": 1564.00,
				"currency": "GBP"
			}
		}, {
			"type": "available",
			"amount": {
				"value": 1564.00,
				"currency": "GBP"
			}
		}],
		"id": "dba193e4-5e1c-40ad-bd53-a4f590db5c72"
	}]
}