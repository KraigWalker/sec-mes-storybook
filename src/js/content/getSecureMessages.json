{
    "meta": {
        "total_no": 26,
        "count_by_status": {
          "NEW": 3,
          "DRAFT": 0,
          "READ": 10,
          "SENT": 13,
          "DELETED": 2
        }
      },
    "securemessages": [
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-43ea-873f-c",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "111111-11111111"
        },
        "payload": {
            "body": {
                "data": "Hi, \nI am so so..."
            }
        },
        "date_created": "2017-10-17T10:56:43",
        "status": "PENDING"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-43ea-873f-c",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "222222-22222222"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-10-19T10:56:43.511Z",
        "status": "DRAFT"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-43ea-873f-c",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "333333-33333333"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-01-17T10:56:43.511Z",
        "status": "SENT"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-43ea-873f-c",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "444444-44444444"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-09-17T10:56:43.511Z",
        "status": "NEW"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-43ea-873f-c",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "555555-55555555"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-10-01T10:56:43.511Z",
        "status": "SENT"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-43ea-873f-c",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "666666-66666666"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-10-20T10:56:43.511Z",
        "status": "NEW"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-43ea-873f-c",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "777777-77777777"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-10-01T10:56:43.511Z",
        "status": "SENT"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-c75ff61c2b4b",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "888888-88888888"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-10-17T10:56:43.511Z",
        "status": "SENT"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-c75ff61c2b4b",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "999999-99999999"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-10-02T10:56:43.511Z",
        "status": "SENT"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-c75ff61c2b4b",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "000000-00000000"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2016-10-09T10:56:43.511Z",
        "status": "PENDING"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-c75ff61c2b4b",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "000001-00000001"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-10-15T10:56:43.511Z",
        "status": "SENT"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-c75ff61c2b4b",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "000002-00000002"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-10-05T10:56:43.511Z",
        "status": "SENT"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-5cf1-c75ff61c2b4b",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "000003-00000003"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-04-09T10:56:43.511Z",
        "status": "SENT"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-c75ff61c2b4b",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "000004-00000004"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-05-17T10:56:43.511Z",
        "status": "PENDING"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-c75ff61c2b4b",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "000005-00000005"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-08-17T10:56:43.511Z",
        "status": "SENT"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-c75ff61c2b4b",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "000006-00000006"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-09-01T10:56:43.511Z",
        "status": "READ"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-c75ff61c2b4b",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "000007-00000007"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-08-01T10:56:43.511Z",
        "status": "SENT"
    },
    {
        "id": "f7094e0f-bd03-465e-95f0-c89dd168c121",
        "thread_id": "24569088-c75ff61c2b4b",
        "reference": "L09SDF09SDFSD",
        "subject": "Payment Enquiries",
        "account": {
            "id": "$UUID",
            "number": "000008-00000008"
        },
        "payload": {
            "body": {
                "data": "A message sent by a customer to the bank"
            }
        },
        "date_created": "2017-01-23T10:56:43.511Z",
        "status": "PENDING"
    }
]
}