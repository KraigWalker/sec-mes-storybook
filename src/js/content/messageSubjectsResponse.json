{
    "subjects": [
        {
          "key": "BN",
          "value": "Borrowing Needs"
        },
        {
          "key": "PE",
          "value": "Payment Enquiry"
        },
        {
          "key": "GF",
          "value": "General Feedback"
        },
        {
          "key": "TS",
          "value": "Technical Support"
        },
        {
          "key": "OE",
          "value": "Overdraft Enquiries"
        },
        {
          "key": "CA",
          "value": "Change of address enquiries"
        },
        {
          "key": "CO",
          "value": "Complaints"
        }
      ]
}