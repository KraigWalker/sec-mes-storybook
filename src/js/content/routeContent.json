[
    {
        "headerDetails" : {
            "title": "Secure messages",
            "subtext": ["<p>Got an enquiry? Send a message to the Team at B - we’re here to help. What’s more, because you’re logged in, these messages are safe and secure.</p>",
                        "<p>Messages are displayed going back 13 months. For more info, please <a href='#'>search the FAQs</a> about message history.</p>"]
        },
        "path": "/securemessages",
        "component" : "securemessages"
    },
    {
        "headerDetails" : {
            "inboxtitle": "Inbox message",
            "senttitle": "Sent message"
        },
        "path": "/viewmessage",
        "component" : "newmessage"
    },
    {
        "headerDetails" : {
            "title": "New message"
        },
        "path": "/newsecuremessage",
        "component" : "newsecuremessage"
    }

]