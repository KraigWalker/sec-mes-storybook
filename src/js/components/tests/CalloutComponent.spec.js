import React from 'react';
import content from '../../content';
import { shallow, mount } from 'enzyme';
import TestData from '../../content/secureMessagesTestData.json'
import CalloutComponent from '../common/CalloutComponent'
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe('CalloutComponent Component Check', () => {

  const testData = { "accessibilityMessage": 'this is check' }
  const componentWrap = mount(<CalloutComponent dClass='callout' paraText='sampletext' />);

  it("LandingPage component should mount", () => {
    expect(typeof componentWrap).toBe('object');
  });
  it("It should be visually hidden", () => {
    expect(componentWrap.html()).toBe('<div class="callout"><p class="">sampletext</p></div>');
  });
  it("Props Check", () => {
    expect(componentWrap.props().paraText).toEqual('sampletext');
    expect(componentWrap.props().dClass).toEqual('callout');
  });
  it('should render correctly', () => {
    const tree = shallow(
      <CalloutComponent />
    );
    expect(tree).toMatchSnapshot();
  });
});
