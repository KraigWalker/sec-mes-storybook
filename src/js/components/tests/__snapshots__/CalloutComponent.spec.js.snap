// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalloutComponent Component Check should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CalloutComponent
    dClass="callout"
    pClass=""
    paraText="Contrary to popular belief, Lorem Ipsum is not random text. It has roots in a piece of classical Latin literature from 45 BC, making it 2000 years old."
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className=""
      >
        Contrary to popular belief, Lorem Ipsum is not random text. It has roots in a piece of classical Latin literature from 45 BC, making it 2000 years old.
      </p>,
      "className": "callout",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Contrary to popular belief, Lorem Ipsum is not random text. It has roots in a piece of classical Latin literature from 45 BC, making it 2000 years old.",
        "className": "",
      },
      "ref": null,
      "rendered": "Contrary to popular belief, Lorem Ipsum is not random text. It has roots in a piece of classical Latin literature from 45 BC, making it 2000 years old.",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className=""
        >
          Contrary to popular belief, Lorem Ipsum is not random text. It has roots in a piece of classical Latin literature from 45 BC, making it 2000 years old.
        </p>,
        "className": "callout",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Contrary to popular belief, Lorem Ipsum is not random text. It has roots in a piece of classical Latin literature from 45 BC, making it 2000 years old.",
          "className": "",
        },
        "ref": null,
        "rendered": "Contrary to popular belief, Lorem Ipsum is not random text. It has roots in a piece of classical Latin literature from 45 BC, making it 2000 years old.",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
